Corona-Warn-App/src/test/java/de/rki/coronawarnapp/datadonation/analytics/modules/clientmetadata/ClientVersionParserTest.kt

```diff
@@ -0,0 +1,35 @@
+package de.rki.coronawarnapp.datadonation.analytics.modules.clientmetadata
+
+import de.rki.coronawarnapp.BuildConfig
+import io.kotest.matchers.shouldBe
+import io.mockk.MockKAnnotations
+import io.mockk.clearAllMocks
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import testhelpers.BaseTest
+
+class ClientVersionParserTest : BaseTest() {
+
+    @BeforeEach
+    fun setup() {
+        MockKAnnotations.init(this)
+    }
+
+    @AfterEach
+    fun tearDown() {
+        clearAllMocks()
+    }
+
+    private fun createInstance() = ClientVersionParser()
+
+    @Test
+    fun `client version is properly parsed`() {
+        val currentVersionName = BuildConfig.VERSION_NAME
+        val currentVersionNumber = BuildConfig.VERSION_CODE
+
+        val parsedVersion = createInstance().parseClientVersion(currentVersionNumber)
+
+        parsedVersion.toVersionString() shouldBe currentVersionName
```

Comment:
Careful. Only `deviceForTesters` Variant has `-RCx`, would this unit test fail  if executed for a different build flavor?

Reply:
noted, tossed out the whole parsing of version codes and switched to major minor and patch version being provided by the buildconfig